### ISSUE TITLE
Cart items with wrong discount after a coupon for only one product is applied

### ISSUE THREAD

**Issue by crispossa** [co-cart/co-cart#96]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of CoCart on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue.
- [ ] I have included a failing test as a pull request (Optional)
- [x] I have installed the requirements to run this plugin.

## Steps to reproduce the issue

1. On a Woocommerce store with at least 2 products, create a coupon  limited to only one of them products with a fixed discount of less than the value of the product. Eg: product has price of U$10 and fixed discount is U$ 5.
2. Through Postman or any other API testing tool, add both products to the cart
3. Apply the coupon created on step 1.
4. Run the get-cart route.

## Expected/actual behaviour

When I follow those steps, I see both products being discounts by U$ 2.5 each, even the other product than the specified in coupon settings.

I was expecting to see the discount applied to only the right product, not affecting the discount of the other product.

## Isolating the problem

- [ ] This bug happens with only WooCommerce and CoCart plugin are active.
- [x] This bug happens with a default WordPress theme active.
- [ ] This bug happens with the WordPress theme Storefront active.
- [x] This bug happens with the latest release of WooCommerce active.
- [x] I can reproduce this bug consistently using the steps above.

**Comment by seb86** [comment]
@crispossa Have you tried testing the coupon on the front end of your site and not via Postman? Do you get the same results?

Also, you did not tick only happens with WooCommerce and CoCart active. Would you mind sharing your system status report please?

**Comment by crispossa** [comment]
I tried on the front-end of my site first, that's how I realized this bug.

Have a try:

Access: www.labiexames.com.br
Add the two products to cart:
'Hemograma Completo' and 'Colesterol Total'
Apply the coupon 'testecris' to the cart. (this coupon has $5 of fixed discount and is valid only for Hemograma Completo)
See the values of both products being decreased by $2.5 each.

The expected behavior is only Hemograma Completo bedecreased by $5.

I'm a developer and CoCart Pro custommer, but registered with another e-mail and in behalf of a company (labiexames).
